xtask/codegen/css.ungram

```diff
@@ -459,6 +460,154 @@ CssFontFaceAtRule =
 	'font-face'
 	block: CssBlock
 
+
+// <container-condition> = [ <container-name> ]? <container-query>
+// <container-name> = <custom-ident>
+// <container-query>     = not <query-in-parens>
+//                      | <query-in-parens> [ [ and <query-in-parens> ]* | [ or <query-in-parens> ]* ]
+//
+// <query-in-parens>     = ( <container-query> )
+//                       | ( <size-feature> )
+//                       | style( <style-query> )
+//                       | <general-enclosed>
+//
+// <style-query>         = not <style-in-parens>
+//                       | <style-in-parens> [ [ and <style-in-parens> ]* | [ or <style-in-parens> ]* ]
+//                       | <style-feature>
+//
+// <style-in-parens>     = ( <style-query> )
+//                       | ( <style-feature> )
+//                       | <general-enclosed>
+
+// <container-condition> = [ <container-name> ]? <container-query>
+CssContainerAtRule =
+	'container'
+	name: CssIdentifier?
+	query: AnyCssContainerQuery
+	block: CssBlock
+
+// <container-query>
+AnyCssContainerQuery =
+	CssContainerNotQuery
+	| AnyCssContainerCombinableQuery
+
+AnyCssContainerCombinableQuery =
+	CssContainerAndQuery
+	| CssContainerOrQuery
+	| AnyCssContainerQueryInParens
+
+CssContainerNotQuery =
+	'not'
+	query: AnyCssContainerQueryInParens
+
+CssContainerAndQuery =
+	left: AnyCssContainerQueryInParens
+	'and'
+	right: AnyCssContainerCombinableQuery
+
+CssContainerOrQuery =
+	left: AnyCssContainerQueryInParens
+	'or'
+	right: AnyCssContainerCombinableQuery
+
+// <query-in-parens>
+AnyCssContainerQueryInParens =
+	CssContainerQueryInParens
+	| CssContainerSizeFeatureInParens
+	| CssContainerStyleQueryInParens
+
+// ( <container-query> )
+CssContainerQueryInParens =
+	'('
+	query: AnyCssContainerQuery
+	')'
+
+// ( <size-feature> )
+CssContainerSizeFeatureInParens =
+	'('
+	query: AnyCssContainerSizeFeature
+	')'
+
+AnyCssContainerSizeFeature =
```

Comment:
I deliberately avoided reusing the node from the media rule: 
https://drafts.csswg.org/css-contain-3/#typedef-size-feature

_The syntax of a [<size-feature>](https://drafts.csswg.org/css-contain-3/#typedef-size-feature) is the same as for a [media feature](https://drafts.csswg.org/mediaqueries-5/#media-feature)_

But I guess it's better to have different nodes.

Reply:
I am conflicted here, to be honest. The query system and parsing are **the same**, and having two different nodes could *hint* that they are different. I lean more toward having the same nodes.

Reply:
I see your point, I agree with that.
Let's implement the media at rule and see if we can use the same nodes for a size feature.